pkg/autodiscovery/scheduler/meta.go

```diff
@@ -24,28 +27,47 @@ func NewMetaScheduler() *MetaScheduler {
 
 // Register a scheduler in the meta scheduler to dispatch to
 func (ms *MetaScheduler) Register(name string, s Scheduler) {
+	ms.Lock()
+	defer ms.Unlock()
 	if _, ok := ms.activeSchedulers[name]; ok {
 		log.Warnf("Scheduler %s already registered, overriding it", name)
 	}
 	ms.activeSchedulers[name] = s
 }
 
+// Deregister a scheduler in the meta scheduler to dispatch to
+func (ms *MetaScheduler) Deregister(name string) {
+	ms.Lock()
+	defer ms.Unlock()
+	if _, ok := ms.activeSchedulers[name]; !ok {
+		log.Warnf("Scheduler %s no registered, skipping", name)
+		return
+	}
+	delete(ms.activeSchedulers, name)
+}
+
 // Schedule schedules configs to all registered schedulers
 func (ms *MetaScheduler) Schedule(configs []integration.Config) {
+	ms.Lock()
```

Comment:
This will allow types from outside of the package to call `MetaScheduler.Lock()`. I think it should stay as a private variable.